runtime/drivers/snowflake/snowflake.go

```diff
@@ -83,43 +83,78 @@ type configProperties struct {
 	Extras             map[string]any `mapstructure:",remain"`
 }
 
-func (cp configProperties) resolveDSN() (string, error) {
-	if cp.DSN != "" {
-		return cp.DSN, nil
+func (c *configProperties) validate() error {
+	if c.DSN != "" && (c.Account != "" || c.User != "" || c.Database != "" || c.Password != "" || c.Schema != "" || c.Warehouse != "" || c.Role != "" || c.Authenticator != "" || c.PrivateKey != "") {
+		var set []string
+		if c.Account != "" {
+			set = append(set, "account")
+		}
+		if c.User != "" {
+			set = append(set, "user")
+		}
```

Comment:
Same comment as above – each property is checked twice, consider if can be simplified